src/api/warnings.tsx

```diff
@@ -1,14 +1,35 @@
 import { handleResponse } from "util/helpers";
 import type { LxdWarning } from "types/warning";
 import type { LxdApiResponse } from "types/apiResponse";
+import { withEntitlementsQuery } from "util/entitlements/api";
+
+export const warningEntitlements = ["can_delete"];
+
+export const fetchWarnings = async (
+  isFineGrained: boolean | null,
+): Promise<LxdWarning[]> => {
+  const entitlements = withEntitlementsQuery(
+    isFineGrained,
+    warningEntitlements,
+  );
 
-export const fetchWarnings = async (): Promise<LxdWarning[]> => {
   return new Promise((resolve, reject) => {
-    fetch("/1.0/warnings?recursion=1")
+    fetch(`/1.0/warnings?recursion=1${entitlements}`)
       .then(handleResponse)
       .then((data: LxdApiResponse<LxdWarning[]>) => {
         resolve(data.metadata);
       })
       .catch(reject);
   });
 };
+
+export const deleteWarning = async (warning: LxdWarning): Promise<void> => {
+  return new Promise((resolve, reject) => {
+    fetch(`/1.0/warnings/${warning.uuid}`, {
+      method: "DELETE",
+    })
+      .then(handleResponse)
+      .then(resolve)
+      .catch(reject);
+  });
+};
```

Comment:
I think we can remove the wrapped promise since it is an async function. 
```suggestion
export const deleteWarning = async (warning: LxdWarning): Promise<void> => {
  const response = await fetch(`/1.0/warnings/${warning.uuid}`, {
    method: "DELETE",
  });
  await handleResponse(response);
};
```